src/clp_ffi_js/ir/IrStreamReader.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef CLP_FFI_JS_IR_IR_STREAM_READER_HPP
+#define CLP_FFI_JS_IR_IR_STREAM_READER_HPP
+
+#include <Array.hpp>
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <streaming_compression/zstd/Decompressor.hpp>
+#include <vector>
+
+#include <clp/ir/types.hpp>
+#include <clp/TimestampPattern.hpp>
+#include <emscripten/val.h>
+
+#include <clp_ffi_js/ir/LogEventWithLevel.hpp>
+#include <clp_ffi_js/ir/StreamReader.hpp>
+#include <clp_ffi_js/ir/StreamReaderDataContext.hpp>
+
+namespace clp_ffi_js::ir {
+using clp::ir::four_byte_encoded_variable_t;
+
+/**
+ * Mapping between an index in the filtered log events collection to an index in the unfiltered
+ * log events collection.
+ */
+using FilteredLogEventsMap = std::optional<std::vector<size_t>>;
+
+/**
+ * Class to deserialize and decode Zstandard-compressed CLP IRv1 streams as well as format decoded
+ * log events.
+ */
+class IrStreamReader : public StreamReader {
+    friend StreamReader;
+
+public:
+    // Destructor
+    ~IrStreamReader() override = default;
+
+    // Disable copy constructor and assignment operator
+    IrStreamReader(IrStreamReader const&) = delete;
+    auto operator=(IrStreamReader const&) -> IrStreamReader& = delete;
+
+    // Define default move constructor
+    IrStreamReader(IrStreamReader&&) = default;
+    // Delete move assignment operator since it's also disabled in `clp::ir::LogEventDeserializer`.
+    auto operator=(IrStreamReader&&) -> IrStreamReader& = delete;
```

Comment:
There were some CLP code issues which prevented us from deleting the move constructor with LLVM toolchains (with GCC it could be done). I assume the code issues have been now addressed.
```suggestion
    // Disable move constructor and assignment operator
    IrStreamReader(IrStreamReader&&) = delete;
    auto operator=(IrStreamReader&&) -> IrStreamReader& = delete;
```

Reply:
Not yet, unfortunately. I have the fix on a branch, but it's part of a larger change that needs to be split up and was waiting for some other PRs to get merged. It should end up being merged sometime this week.